internal/api/grpc/policy/v2/policy.go

```diff
@@ -0,0 +1,305 @@
+		SmtpSenderAddressMatchesInstanceDomain: current.SMTPSenderAddressMatchesInstanceDomain,
+	}
+}
+
+func (s *Server) GetLegalSettings(ctx context.Context, req *policy.GetLegalSettingsRequest) (*policy.GetLegalSettingsResponse, error) {
+	orgID := req.GetOrganisation().GetOrgId()
+	if orgID == "" {
+		orgID = authz.GetCtxData(ctx).OrgID
+	}
+
+	current, err := s.query.PrivacyPolicyByOrg(ctx, true, orgID, false)
+	if err != nil {
+		return nil, err
+	}
+	return &policy.GetLegalSettingsResponse{
+		Settings: ModelLegalSettingsToPb(current),
+		Details: &object.Details{
+			Sequence:          current.Sequence,
+			ChangeDate:        timestamppb.New(current.ChangeDate),
+			ResourceOwner:     current.ResourceOwner,
+			ResourceOwnerType: IsDefaultToResourceOwnerTypePb(current.IsDefault),
+		},
+	}, nil
+}
+
+func ModelLegalSettingsToPb(current *query.PrivacyPolicy) *policy.LegalSettings {
+	return &policy.LegalSettings{
+		TosLink:      current.TOSLink,
+		PrivacyLink:  current.PrivacyLink,
+		HelpLink:     current.HelpLink,
+		SupportEmail: string(current.SupportEmail),
+	}
+}
+
+func (s *Server) GetLockoutPolicy(ctx context.Context, req *policy.GetLockoutPolicyRequest) (*policy.GetLockoutPolicyResponse, error) {
+	orgID := req.GetOrganisation().GetOrgId()
+	if orgID == "" {
+		orgID = authz.GetCtxData(ctx).OrgID
+	}
+
+	current, err := s.query.LockoutPolicyByOrg(ctx, true, orgID, false)
+	if err != nil {
+		return nil, err
+	}
+	return &policy.GetLockoutPolicyResponse{
+		Policy: ModelLockoutPolicyToPb(current),
+		Details: &object.Details{
+			Sequence:          current.Sequence,
+			ChangeDate:        timestamppb.New(current.ChangeDate),
+			ResourceOwner:     current.ResourceOwner,
+			ResourceOwnerType: IsDefaultToResourceOwnerTypePb(current.IsDefault),
+		},
+	}, nil
+}
+
+func ModelLockoutPolicyToPb(current *query.LockoutPolicy) *policy.LockoutPolicy {
+	return &policy.LockoutPolicy{
+		MaxPasswordAttempts: current.MaxPasswordAttempts,
+	}
+}
+
+func (s *Server) GetActiveIdentityProviders(ctx context.Context, req *policy.GetActiveIdentityProvidersRequest) (*policy.GetActiveIdentityProvidersResponse, error) {
+	orgID := req.GetOrganisation().GetOrgId()
+	if orgID == "" {
+		orgID = authz.GetCtxData(ctx).OrgID
+	}
+
+	links, err := s.query.IDPLoginPolicyLinks(ctx, orgID, &query.IDPLoginPolicyLinksSearchQuery{}, false)
+	if err != nil {
+		return nil, err
+	}
+
+	idps := make([]string, len(links.Links))
+	for i, d := range links.Links {
+		idps[i] = d.IDPID
+	}
+	return &policy.GetActiveIdentityProvidersResponse{
+		Idps:    idps,
+		Details: &object.Details{},
+	}, nil
+}
+
+func (s *Server) GetGeneralSettings(ctx context.Context, _ *policy.GetGeneralSettingsRequest) (*policy.GetGeneralSettingsResponse, error) {
+	langs, err := s.query.Languages(ctx)
+	if err != nil {
+		return nil, err
+	}
+	instance, err := s.query.Instance(ctx, true)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
the instance (incl. default org id, default lang) can be directly retrieved from the context: authz.GetInstance(ctx)